src/invidious.cr

```diff
@@ -1831,373 +1833,6 @@ end
-
-        text = HTML.unescape(node.content)
-        text = text.gsub(/<font color="#[a-fA-F0-9]{6}">/, "")
-        text = text.gsub(/<\/font>/, "")
-        if md = text.match(/(?<name>.*) : (?<text>.*)/)
-          text = "<v #{md["name"]}>#{md["text"]}</v>"
-        end
-
-        str << <<-END_CUE
-        #{start_time} --> #{end_time}
-        #{text}
-
-
-        END_CUE
-      end
-    end
-  else
-    webvtt = YT_POOL.client &.get("#{url}&format=vtt").body
-  end
-
-  if title = env.params.query["title"]?
-    # https://blog.fastmail.com/2011/06/24/download-non-english-filenames/
-    env.response.headers["Content-Disposition"] = "attachment; filename=\"#{URI.encode_www_form(title)}\"; filename*=UTF-8''#{URI.encode_www_form(title)}"
-  end
-
-  webvtt
-end
-
-get "/api/v1/comments/:id" do |env|
-  locale = LOCALES[env.get("preferences").as(Preferences).locale]?
-  region = env.params.query["region"]?
-
-  env.response.content_type = "application/json"
-
-  id = env.params.url["id"]
-
-  source = env.params.query["source"]?
-  source ||= "youtube"
-
-  thin_mode = env.params.query["thin_mode"]?
-  thin_mode = thin_mode == "true"
-
-  format = env.params.query["format"]?
-  format ||= "json"
-
-  action = env.params.query["action"]?
-  action ||= "action_get_comments"
-
-  continuation = env.params.query["continuation"]?
-  sort_by = env.params.query["sort_by"]?.try &.downcase
-
-  if source == "youtube"
-    sort_by ||= "top"
-
-    begin
-      comments = fetch_youtube_comments(id, PG_DB, continuation, format, locale, thin_mode, region, sort_by: sort_by, action: action)
-    rescue ex
-      next error_json(500, ex)
-    end
-
-    next comments
-  elsif source == "reddit"
-    sort_by ||= "confidence"
-
-    begin
-      comments, reddit_thread = fetch_reddit_comments(id, sort_by: sort_by)
-      content_html = template_reddit_comments(comments, locale)
-
-      content_html = fill_links(content_html, "https", "www.reddit.com")
-      content_html = replace_links(content_html)
-    rescue ex
-      comments = nil
-      reddit_thread = nil
-      content_html = ""
-    end
-
-    if !reddit_thread || !comments
-      env.response.status_code = 404
-      next
-    end
-
-    if format == "json"
-      reddit_thread = JSON.parse(reddit_thread.to_json).as_h
-      reddit_thread["comments"] = JSON.parse(comments.to_json)
-
-      next reddit_thread.to_json
-    else
-      response = {
-        "title"       => reddit_thread.title,
-        "permalink"   => reddit_thread.permalink,
-        "contentHtml" => content_html,
-      }
-
-      next response.to_json
-    end
-  end
-end
-
-get "/api/v1/insights/:id" do |env|
-  locale = LOCALES[env.get("preferences").as(Preferences).locale]?
-  next error_json(410, "YouTube has removed publicly available analytics.")
-end
```

Comment:
You should'nt remove this endpoint. This is part of the v1 API, and imo we should keep it consistent, even if youtube removed the feature.

Reply:
Eh I disagree. I mean Invidious currently works off of a rolling release model and I don't see why the API has to be different  —  anything that is planned to be removed would obviously have a deprecation notice as well.

In the case of the insights endpoint, it has been deprecated for almost 2 years now. See https://github.com/iv-org/invidious/commit/aba2c5b9387a8968ac1e49fe181e7bc6ca4e3b1b. I just don't see the point of still keeping it within the code base.